subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/horreum_lts_store.py

```diff
@@ -125,3 +126,89 @@ def build_lts_payloads() -> dict:
+        RESULTS = entry.results
+        
+        start_time: datetime.datetime = RESULTS.start_time
+        end_time: datetime.datetime = RESULTS.end_time
+
+        output = {
+            "$schema": "urn:rhods-matbench-upload:3.0.0",
+            "data": {
+                "users": _decode_limited_users(RESULTS.ods_ci, RESULTS.testpod_hostnames, RESULTS.notebook_pod_times),
+                'rhods_version': RESULTS.rhods_info.version,
+                'ocp_version': RESULTS.sutest_ocp_version,
+                'metrics': _gather_prom_metrics(RESULTS.metrics),
+                'thresholds': RESULTS.thresholds,
+                'config': RESULTS.test_config.yaml_file
+            },
+            "metadata": {
+                "test": "rhods-notebooks-ux",
+                "start": start_time.isoformat(),
+                "end": end_time.isoformat()
+            }
+        }
+        
+        yield output, start_time, end_time
+
+
+def _decode_limited_users(users, hostnames, pod_times):
+    output = []
+    for (key, val) in users.items():
+        output.append({
+            'hostname': hostnames[key],
+            'steps': _decode_limited_steps(val.output, pod_times[key]),
+            'succeeded': val.exit_code == 0
+        })
+
+    return output
+
+
+def _decode_limited_steps(steps, pod_times):
+    out_steps = []
+    for (step_name, step_data) in steps.items():
+        out_step = {
+            'name': step_name,
+            'duration': (step_data.finish - step_data.start).total_seconds(),
+            'status': step_data.status
+        }
+        if step_name in ("Wait for the Notebook Spawn", "Create and Start the Workbench"):
+            out_step['substeps'] = _generate_pod_timings(pod_times, step_data.start, step_data.finish)
+        
+        out_steps.append(out_step)
+
+    return out_steps
+
+def _generate_pod_timings(pod_times, start, end):
+    output = {}
+
+    if hasattr(pod_times, "pod_scheduled"):
+        output['resource_init_time'] = (pod_times.pod_scheduled - start).total_seconds()
+    if hasattr(pod_times, "containers_ready"):
+        output['container_ready_time'] = (pod_times.containers_ready - pod_times.pod_initialized).total_seconds()
+    if hasattr(pod_times, 'containers_ready'):
+        output['user_notification'] = (end - pod_times.containers_ready).total_seconds()
+    
+    return output
+
+def _gather_prom_metrics(metrics) -> dict:
+    out = {}
+    prom_metrics = {
+        "sutest": [
+            "Sutest API Server Requests (server errors)",
+            "Sutest Control Plane Node CPU idle",
+            "sutest__container_cpu__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard",
+            "sutest__container_cpu_requests__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard",
+            "sutest__container_cpu_limits__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard"
```

Comment:
can you try to integrate these definition in `plotting/prom.py`?

something like:

```
def _get_apiserver_errcodes(cluster_role, register):
    all_metrics = []
    server_errors_metric_name = "{cluster_role.title()} API Server Requests (server errors)"
    apiserver_request_metrics = [
        {f"{cluster_role.title()} API Server Requests (successes)": 'sum by (code) (increase(apiserver_request_total{code=~"2.."}[2m]))'},
        {f"{cluster_role.title()} API Server Requests (client errors)": 'sum by (code) (increase(apiserver_request_total{code=~"4.."}[2m]))'},
        {server_errors_metric_name: 'sum by (code) (increase(apiserver_request_total{code=~"5.."}[2m]))'},
    ]
    
    if register: 
      if cluster_role == "sutest": register_lts_metric(cluster_role, server_errors_metric_name)
```
and in `register_lts_metric` you populate a list, and in this file (horreum_lts_store.py) you just traverse this list
the function above (`_get_apiserver_errcodes`) might be called multiple times, but only once with `register=True`